### PR TITLE
Login & sign up

### PR DESCRIPTION
**Pull request by aleks279**
Note: during the process of creating the login, hit a problem -> there were 3 different types of `devise` users, so single login was not achievable.

Refactored the DB to have 1 single `devise` user model (aside from the admin) and changed the client, musician and band models to `has_one` user, and the user `belongs_to` band, client and musician.

Attributes for user are the minimum required: email & password. Other particular attributes, such as names and phones, are managed by each individual model.

Steps to recreate the new DB:
- `rake db:drop:all`
- `rake db:create:all`
- `rake db:migrate`
- `rake db:test:prepare` -> for Rspec
- `rake sample_data:all` -> load sample data

### CODE REVIEW

File: lib/tasks/sample_data.rake [hunk 1]
```diff
@@ -16,111 +18,129 @@ namespace :sample_data do
...
+        event_type_id: 1 + rand(EventType.all.count)
       )
     end
   end
 
   def update_or_create_client(attributes)
-    client = Client.find_or_initialize_by(email: attributes.delete(:email))
+    client = Client.find_or_initialize_by(name: attributes.delete(:name))
     client.update_attributes(attributes)
+    return client
```
Comment (jramoscr) [inline lib/tasks/sample_data.rake]: I know you know, but Ruby always returns the last line, unless you were using an explicit early return for some cases, so, in this line the `return` statement is not necessary, unless the team has agreed to use it explicitly.

Comment (aleks279) [inline lib/tasks/sample_data.rake]: As you say my man, I indeed know that Ruby does return the last line and as always, there's a reasonable explanation for my actions :smile: 

The method `update_attributes` you see in line 121 does not return the object it's updating, it returns a `true` value if it succeeds or a `false` value if it fails, that's why it is commonly used like this:

```
if @foo.update_attributes(params)
    # life is good
else
    # something is wrong
end
```

Sources: 
- http://apidock.com/rails/ActiveRecord/Base/update_attributes
- http://apidock.com/rails/ActiveRecord/Persistence/update_attributes

However, if you check lines `#25 #39 #56 #64 #72 #80 #88 #95`, what I need from the methods `update_or_create_band`, `update_or_create_musician` and `update_or_create_client` is not a `true` or `false` value, but the object that was just created/updated, and that my friend, is why I need that `return` statement exactly where it is :grin:

Comment (jramoscr) [inline lib/tasks/sample_data.rake]: Yes, I see that you need the object, but in that case you can type `client` in the last line, and avoid the usage of the `return` statement. In this line, you don't need the statement explicitly. Check this: https://github.com/bbatsov/ruby-style-guide#no-explicit-return

Comment (aleks279) [inline lib/tasks/sample_data.rake]: That link, I like it. And as a bonus, I'll change this to fit the Ruby style guide :)

Comment (jramoscr) [inline lib/tasks/sample_data.rake]: :+1:

**Pull request merged**